types/http-proxy/index.d.ts

```diff
@@ -193,6 +193,8 @@ declare namespace Server {
         selfHandleResponse?: boolean | undefined;
         /** Buffer */
         buffer?: stream.Stream | undefined;
+        /** Explicitly set the method type of the ProxyReq */
+        method?: string | undefined;
```

Comment:
Could be set to literal values `GET | POST | PUT | PATCH | DELETE | OPTIONS` but otherwise looks good

Reply:
Thanks for looking at this so quickly. I put a comment about this above

> I did originally implement the type as an enum of the established HTTP method types, but as the node http 'method' is just typed as string I decided to do the same here.

So basically my thoughts are if node is doing it as string, this should as well because we could end up in a situation where the proxy is less permissive than the target server. Practically this would only happen if someone implemented their own custom http method but who am I to say they can't do that?

LMK if this changes your opinion, otherwise can switch it back to an enum if we're sure.

Reply:
indeed missed that, fair enough 👌🏻